homeassistant/components/imap/coordinator.py

```diff
@@ -77,7 +77,9 @@ async def refresh_email_count(self) -> int:
                 f"Invalid response for search '{self.config_entry.data[CONF_SEARCH]}': {result} / {lines[0]}"
             )
         if self.support_push:
-            self.hass.async_create_task(self.async_wait_server_push())
+            self.hass.async_create_background_task(
```

Comment:
We need to check if a previous background task is running as well so we do not keep creating them each time the update happens.

Reply:
I'll dig a bit more. Something still feels like there is a chance of a race here where we can get multiple tasks or if the coordinator is called to refresh manually?

Reply:
@bdraco your suspicions are valid. With IMAP enabled and updating to 2023.3.5, start up times are normal, however I am still seeing these task was destroyed pending messages in my log every once in a while. I'll have to confirm that with debugpy again, that they are indeed coming from IMAP but it was the only thing that I changed (enabling IMAP) other than updating to 2023.3.5

Reply:
Jan fixed it in a larger refactor https://github.com/home-assistant/core/pull/89759

I haven't had a chance to test it yet though as I've been fighting with https://github.com/home-assistant/core/pull/89824 quite a bit more than I had hoped